utils/s2n_mem.c

```diff
@@ -85,7 +85,11 @@ static int s2n_mem_malloc_mlock_impl(void **ptr, uint32_t requested, uint32_t *a
     /* Page aligned allocation required for mlock */
     uint32_t allocate;
 
+/* Avoids spurious alarms over signed to unsigned conversion in (uint32_t)page_size. */
```

Comment:
Could we just change the types to prevent the problem in the first place?

Reply:
I would actually prefer to change `page_size` to `uint32_t`. The return value from `sysconf` is signed because it can return -1 on errors. Since we're `GUARD`ing against that, it would be better to store it as a `uint32_t` after that, as I would much prefer to check it once than everywhere that it's used.

Reply:
Implemented as suggested.